de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/project/FileSystemChangeListener.java

```diff
@@ -1,529 +1,867 @@
+            }
 
-        if (project == null || !isValidProject(project) ||
-            !isCompletelyShared(project)) {
             return;
         }
 
-        path = makeAbsolutePathProjectRelative(path, project);
+        fireActivity(activity);
 
-        SPath spath = new SPath(project, path);
+        if (oldPathIsShared) {
+            if (fileIsOpen) {
+                EditorActivity closeOldEditorActivity = new EditorActivity(user,
+                    EditorActivity.Type.CLOSED, oldFilePath);
 
-        User user = resourceManager.getSession().getLocalUser();
-        IActivity activity;
+                fireActivity(closeOldEditorActivity);
+            }
 
-        byte[] bytes = new byte[0];
-        try {
-            bytes = virtualFileCopyEvent.getOriginalFile()
-                .contentsToByteArray();
-        } catch (IOException e) {
-            LOG.error("could not read content of original file "
-                + virtualFileCopyEvent.getOriginalFile(), e);
-            return;
+            //TODO reset the vector time for the old file
         }
 
-        activity = new FileActivity(user, Type.CREATED, Purpose.ACTIVITY, spath,
-            null, bytes, virtualFile.getCharset().name());
+        if (newPathIsShared && fileIsOpen) {
+            SPath newFilePath = new SPath(newParentPath.getProject(),
+                newParentPath.getProjectRelativePath().append(relativePath));
 
-        project.addFile(newFile);
+            EditorActivity openNewEditorActivity = new EditorActivity(user,
+                EditorActivity.Type.ACTIVATED, newFilePath);
 
-        resourceManager.internalFireActivity(activity);
+            fireActivity(openNewEditorActivity);
+        }
     }
 
+    /**
+     * {@inheritDoc}
+     * <p></p>
+     * Handles name changes for resource as resource moves. Generates and
+     * dispatches the needed activities. For directories, the listener is not
+     * called for all contained resources, meaning these resources are also
+     * handled by the call for the directory.
+     * <p></p>
+     * Other property changes are ignored.
+     *
+     * @param filePropertyEvent {@inheritDoc}
+     * @see #generateFolderMove(VirtualFile, VirtualFile, VirtualFile, String)
+     * @see #generateFileMove(VirtualFile, VirtualFile, VirtualFile, String, String)
+     */
     @Override
     public void beforePropertyChange(
         @NotNull
-        VirtualFilePropertyEvent filePropertyEvent) {
-        // Not interested
+            VirtualFilePropertyEvent filePropertyEvent) {
+
+        assert enabled : "the before property change listener was triggered while it was disabled";
+
+        VirtualFile file = filePropertyEvent.getFile();
+        String propertyName = filePropertyEvent.getPropertyName();
+        Object oldValue = filePropertyEvent.getOldValue();
+        Object newValue = filePropertyEvent.getNewValue();
+
+        switch (propertyName) {
```

Comment:
Remnant of an old implementation that was also checking whether the writable status of a file changed. Since I did not know what to do with this information, I removed this second case. I don't think this switch hurts. It makes it easier to add new cases if they are needed later on. (Other options are defined in in the class [VirtualFile](https://github.com/JetBrains/intellij-community/blob/6c276a28c71641322155d155470ee1b1d4f9843c/platform/core-api/src/com/intellij/openapi/vfs/VirtualFile.java#L54).)